upgrade/util/util.go

```diff
@@ -149,6 +159,99 @@ func CreateOrUpdateLonghornVersionSetting(namespace string, lhClient *lhclientse
 	return nil
 }
 
+// CheckUpgradePathSupported returns if the upgrade path from lhCurrentVersion to meta.Version is supported.
+//
+//	For example: upgrade path is from x.y.z to a.b.c,
+//	0 <= a-x <= 1 is supported, and y should be after a specific version if a-x == 1
+//	0 <= b-y <= 1 is supported when a-x == 0
+//	all downgrade is not supported
+func CheckUpgradePathSupported(namespace string, lhClient lhclientset.Interface) error {
```

Comment:
Please add more one path is not allowed. Any pre-GA version upgrade is not allowed. For example, `1.4.2-rc1` to `1.5.0`.  You should be able to use semver function to achieve the comparison check. 

error log should highlight any non-GA version is not supported.

Reply:
Added the check statement for non-GA version across one minor version and error log.
Do we support non-GA version upgrade e.g., from v1.5.0-rc1 to v1.5.0-rc2 and across patch versions e.g. from v1.5.0-rc3 to v1.5.0 or v1.5.1?
I assume we support these scenarios.

Reply:
Let's remove the non-GA version check first because it would impact our internal testing for the upgrade. 

I will create another ticket to tackle this in 1.6.

Reply:
Removed the non-GA version check.